### PR TITLE
Proposed solution to issue #404

### PR DESCRIPTION
**Pull request by tiagogodinho**
This is a proposed solution to issue #404.

The problem:

``` ruby
task = Task.find(1)
# => #<Task(tasks/1) title="Test" created_at="2016-08-04T20:03:42.612Z" updated_at="2016-08-06T17:00:35.649Z" id=1>
task.changes
# => {}

task = Task.all[0]
# => #<Task(tasks/1) title="Test" created_at="2016-08-04T20:03:42.612Z" updated_at="2016-08-06T17:00:35.649Z" id=1>
task.changes
# => {"title"=>[nil, "Test"], "created_at"=>[nil, "2016-08-04T20:03:42.612Z"], "updated_at"=>[nil, "2016-08-06T17:00:35.649Z"]} 
```

Now:

``` ruby
task = Task.find(1)
task.changes
# => {}

task = Task.all[0]
task.changes
# => {}
task.title = 'New title'
task.changes
# => {"title"=>["Test", "New title"]}
```

### CODE REVIEW

**Comment by lucasrenan** [comment]
👍

**Comment by edtjones** [comment]
hi @tiagogodinho @lucasrenan  I've just got commit access on this repo and plan to get back on track with maintenance over time - will be in touch soon!

File: spec/model/dirty_spec.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@
         builder.use Her::Middleware::FirstLevelParseJSON
         builder.use Faraday::Request::UrlEncoded
         builder.adapter :test do |stub|
+          stub.get("/users") { |env| [200, {}, [{ :id => 1, :fullname => "Lindsay Fünke" }].to_json] }
```
Comment (zacharywelch) [inline spec/model/dirty_spec.rb]: Can we add multiple users here for [consistency](https://github.com/remiprev/her/blob/master/spec/model/orm_spec.rb#L13)?

File: spec/model/dirty_spec.rb [hunk 2]
```diff
@@ -25,6 +26,14 @@
       end
     end
 
+    context "for existing resource in a collection" do
```
Comment (zacharywelch) [inline spec/model/dirty_spec.rb]: how about?
```ruby
context "for existing collections" do
  let(:users) { Foo::User.all }
  it "has no changes" do
    expect(users).to_not include(be_changed)
  end
end
```

File: lib/her/model/attributes.rb [hunk 3]
```diff
@@ -36,14 +36,23 @@ def self.initialize_collection(klass, parsed_data={})
...
           end
           resource
         end
         Her::Collection.new(collection_data, parsed_data[:metadata], parsed_data[:errors])
       end
 
+      # Initialize a resource
+      #
+      # @private
+      def self.initialize_resource(klass, item_data)
```
Comment (zacharywelch) [inline lib/her/model/attributes.rb]: this should come in handy elsewhere 👍

**Comment by zacharywelch** [comment]
Fixed by #298. Thanks for your help on this @tiagogodinho!

**Pull request closed without merging**